File: managment_tournoi_backend/src/main/java/ma/youcode/managment_tournoi_backend/dto/matchDto/MatchShowDto.java
package ma.youcode.managment_tournoi_backend.dto.matchDto;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.FetchType;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ma.youcode.managment_tournoi_backend.dto.appUserFileDto.getDto.MemberShowDto;
import ma.youcode.managment_tournoi_backend.dto.matchTeamDto.MatchTeamShowDto;
import ma.youcode.managment_tournoi_backend.entity.AppUser;
import ma.youcode.managment_tournoi_backend.entity.MatchTeam;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchShowDto {
    private UUID codeMatch;
    private LocalDate date;
    private LocalTime startDateMatch;
    private LocalTime endDateMatch;
    private List<MatchTeamShowDto> matchTeams;
    private MemberShowDto arbitrator;
}

File: managment_tournoi_frontend/src/app/dto/matchDto/matchShowDto.ts
import { MemberShowDto } from "../appUserFileDto/getDto/memberShowDto";
import { MatchTeamShowDto } from "../matchTeamDto/matchTeamShowDto";

export interface MatchShowDto {
    codeMatch: string;
    date: string;
    startDateMatch: string; 
    endDateMatch: string; 
    matchTeams: MatchTeamShowDto[];
    arbitrator: MemberShowDto;
}

